src/elem.rs

```diff
@@ -16,9 +16,20 @@ macro_rules! bin_operation {
             type Output = Elem;
             fn $fct_name(self, other: Self::Output) -> Self::Output {
                 match (self, other) {
-                    (Elem::Image(lhs), Elem::Image(rhs)) => Elem::Image(lhs $op rhs),
-                    (Elem::Value(lhs), Elem::Image(rhs)) => Elem::Image(rhs $op lhs),
-                    (Elem::Image(lhs), Elem::Value(rhs)) => Elem::Image(lhs $op rhs),
+                    (Elem::Image(mut lhs), Elem::Image(rhs)) => {
+                        par_azip!(mut lhs, rhs in {
+                            *lhs = *lhs $op rhs
+                        });
+                        Elem::Image(lhs)
+                    },
```

Comment:
Hey, cool, || this wasn't complicated at all! Or maybe it was but it doesn't show in the code :)

Did you try making `$op` assignment operators, like `+=` instead of just `+`? That way you could write `*lhs $op rhs`. It's probably not faster, I just wonder if it would make the code somewhat cleaner.